src/storage/interfaces/onchain-zkp-verifier.ts

```diff
@@ -21,4 +25,39 @@ export interface IOnChainZKPVerifier {
     txData: ContractInvokeTransactionData,
     zkProofResponses: ZeroKnowledgeProofResponse[]
   ): Promise<Map<string, ZeroKnowledgeProofResponse>>;
+
+  /**
+   * Submit ZKP Response V2 to OnChainZKPVerifier contract.
+   * @beta
+   * @param {Signer} ethSigner - tx signer
+   * @param {txData} ContractInvokeTransactionData - transaction data
+   * @param {ZeroKnowledgeProofResponse[]} zkProofResponses - zkProofResponses
+   * @returns {Promise<Map<string, ZeroKnowledgeProofResponse[]>>} - map of transaction hash - ZeroKnowledgeProofResponse[]
+   */
+  submitZKPResponseV2(
+    ethSigner: Signer,
+    txData: ContractInvokeTransactionData,
+    zkProofResponses: ZeroKnowledgeProofResponse[]
+  ): Promise<Map<string, ZeroKnowledgeProofResponse[]>>;
+
+  /**
+   * Returns the Map of request id to transaction data args for the ZKP verifier contract submission.
+   * For each request id new transaction data is created.
+   * @param txData
+   * @param zkProofResponses
+   */
+  prepareZKPResponseSubmitV1TxData(
+    txData: ContractInvokeTransactionData,
+    zkProofResponses: ZeroKnowledgeProofResponse[]
+  ): Promise<Map<number, JsonDocumentObjectValue[]>>;
+
+  /**
+   * Returns args for the ZKP verifier contract submission V2 (single tx).
+   * @param txData
+   * @param zkProofResponses
+   */
+  prepareZKPResponseSingleTxData(
```

Comment:
let's stick to prepareZKPResponseSubmitV2TxData (I know I said to used single tx, but I'm suspect we will have another sumbit single tx with different functionality.

on the second though prepareTxData should return tx data as hex for both methods, in that case we will have the same interface and less confusion